src/Microsoft.AspNetCore.Razor.Evolution/BoundAttributeDescriptor.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.AspNetCore.Razor.Evolution
+{
+    /// <summary>
+    /// A metadata class describing a tag helper attribute.
+    /// </summary>
+    public abstract class BoundAttributeDescriptor
+    {
+        protected BoundAttributeDescriptor(string kind)
+        {
+            Kind = kind;
+        }
```

Comment:
Why is `Kind` required here and nothing else? If we're going to make it required we should also null-check it. I'd be fine with just not making it required. Extending this set of classes isn't something you do casually.